File: 792-binary-search/binary-search.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        left,right = 0, len(nums) - 1

        while left <= right:
            mid = (left+right)//2

            if nums[mid] == target: return mid

            if nums[mid] > target: right-=1

            elif nums[mid] < target: left+=1

        
        return -1

        # TC -> O(Log N)
        # SC -> O(1)
        

File: 792-binary-search/binary-search.java
class Solution {
    public int search(int[] nums, int target) {
        int i = 0;
        int j = nums.length - 1;

        while(i <= j){
            int midPoint = (j+i)/2;

            if(nums[midPoint] == target){
                return midPoint;
            }else if(nums[midPoint] < target){
                i+=1;
            }else{
                j-=1;
            }
        }

        return -1;


    }
}

// SC -> O(1)
// TC -> O(Log N)